### PR TITLE
compress test

### PR DESCRIPTION
**Pull request by mosquito**
## Checklist

- [X] I think the code is well written
- [X] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by mosquito** [comment]
Hi @asvetlov this PR just proof the bug.

**Comment by webknjaz** [comment]
Please fill out the PR template: https://github.com/aio-libs/aiohttp/blob/master/.github/PULL_REQUEST_TEMPLATE.md

File: tests/test_client_functional.py [hunk 1]
```diff
@@ -1231,6 +1231,21 @@ async def handler(request):
     resp.close()
 
 
+async def test_GET_DEFLATE(aiohttp_client: Any) -> None:
```
Comment (webknjaz) [inline tests/test_client_functional.py:1234]: We can mark this as expected to fail per https://pganssle-talks.github.io/xfail-lightning/, then merge it (provided other requests are addressed), and work on the fix separately:
```suggestion
@pytest.mark.xfail(reason="https://github.com/aio-libs/aiohttp/pull/5478")
async def test_GET_DEFLATE(aiohttp_client: Any) -> None:
```

**Review by webknjaz — CHANGES_REQUESTED** [review]
This PR mustn't contain any unrelated updates to `vendor/http-parser` submodule. Also, please make the PR title more specific and fill in the template (https://github.com/aio-libs/aiohttp/blob/master/.github/PULL_REQUEST_TEMPLATE.md)

File: tests/test_client_functional.py (hunk 1 above)
Comment (mosquito) [reply tests/test_client_functional.py:1234]: it's fails only with picohttpparser, with python implementation it's works expected

Comment (webknjaz) [reply tests/test_client_functional.py:1234]: Then this should be reflected in the PR description.

**Comment by Dreamsorcerer** [comment]
Superseded by #9108.

**Pull request closed without merging**